homeassistant/components/mqtt/lock.py

```diff
@@ -186,57 +182,58 @@ def _setup_from_config(self, config: ConfigType) -> None:
 
         self._valid_states = [config[state] for state in STATE_CONFIG_KEYS]
 
+    @callback
+    def _message_received(self, msg: ReceiveMessage) -> None:
+        """Handle new lock state messages."""
+        payload = self._value_template(msg.payload)
+        if not payload.strip():  # No output from template, ignore
+            _LOGGER.debug(
+                "Ignoring empty payload '%s' after rendering for topic %s",
+                payload,
+                msg.topic,
+            )
+            return
+        if payload == self._config[CONF_PAYLOAD_RESET]:
+            # Reset the state to `unknown`
+            self._attr_is_locked = None
+        elif payload in self._valid_states:
+            self._attr_is_locked = payload == self._config[CONF_STATE_LOCKED]
+            self._attr_is_locking = payload == self._config[CONF_STATE_LOCKING]
+            self._attr_is_open = payload == self._config[CONF_STATE_OPEN]
+            self._attr_is_opening = payload == self._config[CONF_STATE_OPENING]
+            self._attr_is_unlocking = payload == self._config[CONF_STATE_UNLOCKING]
+            self._attr_is_jammed = payload == self._config[CONF_STATE_JAMMED]
```

Comment:
unrelated to this pr, but Lock really should have implemented this as a enum so these are all mutually exclusive.